appservice/src/deploy/getDeployFsPath.ts

```diff
@@ -0,0 +1,61 @@
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { DialogResponses } from 'vscode-azureextensionui';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+
+/**
+ * Appends the deploySubpath setting if the target path matches the root of a workspace folder
+ * If the targetPath is a sub folder instead of the root, leave the targetPath as-is and assume they want that exact folder used
+ */
+export async function getDeployFsPath(targetPath: string, extensionPrefix: string): Promise<string> {
+    if (vscode.workspace.workspaceFolders) {
+        const workspaceConfiguration: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration(extensionPrefix, vscode.Uri.file(targetPath));
+
+        const deploySubpathSetting: string = 'deploySubpath';
+        const deploySubpath: string | undefined = workspaceConfiguration.get(deploySubpathSetting);
+        if (deploySubpath) {
+            if (vscode.workspace.workspaceFolders.some(f => isPathEqual(f.uri.fsPath, targetPath))) {
+                return path.join(targetPath, deploySubpath);
+            } else {
+                const folder: vscode.WorkspaceFolder | undefined = vscode.workspace.workspaceFolders.find(f => isSubpath(f.uri.fsPath, targetPath));
+                if (folder) {
+                    const fsPathWithSetting: string = path.join(folder.uri.fsPath, deploySubpath);
+                    if (!isPathEqual(fsPathWithSetting, targetPath)) {
+                        const settingKey: string = 'showDeploySubpathWarning';
+                        if (workspaceConfiguration.get<boolean>(settingKey)) {
+                            const selectedFolder: string = path.relative(folder.uri.fsPath, targetPath);
+                            const message: string = localize('mismatchDeployPath', 'Deploying "{0}" instead of selected folder "{1}". Use "{2}.{3}" to change this behavior.', deploySubpath, selectedFolder, extensionPrefix, deploySubpathSetting);
+                            // don't wait
+                            // tslint:disable-next-line:no-floating-promises
+                            ext.ui.showWarningMessage(message, { title: localize('ok', 'OK') }, DialogResponses.dontWarnAgain).then(async (result: vscode.MessageItem) => {
+                                if (result === DialogResponses.dontWarnAgain) {
+                                    await workspaceConfiguration.update(settingKey, false, vscode.ConfigurationTarget.Global);
+                                }
+                            });
+                        }
+                    }
+
+                    return fsPathWithSetting;
+                }
+            }
+        }
+    }
+
+    return targetPath;
+}
+
+function isSubpath(expectedParent: string, expectedChild: string): boolean {
```

Comment:
These should be put in a utils file. That's one of the reasons we have "util" files - so that you can just copy the file over to a new repo when you need it

Reply:
Oh, I figured we didn't want to clutter the utils folder with something that is only being used in one file.  Though I guess if I move workspace related code over, then it makes more sense to transfer this to its own file as well.